### ISSUE TITLE
Ensure an error/exception is caught properly by Testfile: `test_custom_install_4_1_energizedprotection.bats`

### ISSUE THREAD

**Issue by a-t-0** [HiveMinds-EU/Productivity-setup#55]
The `BATS` test in path: `/test/Testfile: test_custom_install_4_1_energizedprotection.bats` currently does not correctly detect the error that is thrown if a file is passed that does not exist. This error is, and should be thrown by the `/src/custom_install_4_energizedprotection.sh` file. 

The function in that file checks if the incoming filepath belongs to an existing file or not. If it does not, it throws an error. That error should be caught by an `assert_failure` in the test, with a passing test, if the function fails. Instead, the test fails if the function fails. I also tried throwing an error/exception in the function with: `exit 1` but that is not recognized either by the test.

**Comment by a-t-0** [comment]
So far I tried:
```
#!/bin/bash

# Load bats testing packages
load 'libs/bats-support/load'
load 'libs/bats-assert/load'
load 'libs/bats-file/load'

# Load hardcoded variables and helper functions.
source src/hardcoded_variables.txt
source src/helper.sh
source test/helper.sh

# Create the path to the log file folder.
mkdir -p src/logs

# Method that executes all tested main code before running tests.
setup_file() {
	
	# print test filename to screen.
	if [ "${BATS_TEST_NUMBER}" = 1 ];then
		echo "# Testfile: $(basename "${BATS_TEST_FILENAME}")-" >&3
	fi
	
	# Declare filenames of scripts needed to test this step of the energized protection installer.
	declare -a script_names=("custom_install_4_energizedprotection")
	
	# Specify an additional array with arguments
	declare -a additional_arguments=("test/testfiles/nonexistant_filename.txt")

	# Execute the files and commands needed to test this step of the energized protection installer.
	for i in "${!script_names[@]}"; do
		run_main_functions "${script_names[i]}" "${additional_arguments[i]}"	
	done
}

#---------------------------------------------------------------------------------------------------------------------------
@test "Tests whether an exception is thrown when computing the total nr of lines, if the file is not found." {
	ACTUAL_RESULT=$(<$TOTAL_NR_OF_LINES_IN_TARGET_FILEPATH)
	EXPECTED_OUTPUT="1"

	assert_failure
	#assert_equal "$ACTUAL_RESULT" "$EXPECTED_OUTPUT"
}

@test "Tests whether an exception is thrown when computing the total nr of lines, if the file is not found ERRROR." {
	run_main_functions "custom_install_4_energizedprotection" "test/testfiles/nonexistant_filename.txt"

	expect_failure
	#assert_equal "$ACTUAL_RESULT" "$EXPECTED_OUTPUT"
}

@test "Tests whether an exception is thrown when computing the total nr of lines, if the file is not found ERRROR." {
	! run_main_functions "custom_install_4_energizedprotection" "test/testfiles/nonexistant_filename.txt"

	expect_failure
	#assert_equal "$ACTUAL_RESULT" "$EXPECTED_OUTPUT"
}

@test "Tests whether an exception is thrown when computing the total nr of lines, if the file is not found ERRROR." {
	run_main_functions "custom_install_4_energizedprotection" "test/testfiles/nonexistant_filename.txt"

	assert_failure
	#assert_equal "$ACTUAL_RESULT" "$EXPECTED_OUTPUT"
}

@test "Tests whether an exception is thrown when computing the total nr of lines, if the file is not found ERRROR." {
	! run_main_functions "custom_install_4_energizedprotection" "test/testfiles/nonexistant_filename.txt"

	assert_failure
	#assert_equal "$ACTUAL_RESULT" "$EXPECTED_OUTPUT"
}

@test "Tests whether an exception is thrown when computing the total nr of lines, if the file is not found ERRROR." {
	run_main_functions "custom_install_4_energizedprotection" "test/testfiles/nonexistant_filename.txt"

	[ $status -eq 1 ]
	#assert_equal "$ACTUAL_RESULT" "$EXPECTED_OUTPUT"
}

@test "Tests whether an exception is thrown when computing the total nr of lines, if the file is not found ERRROR." {
	! run_main_functions "custom_install_4_energizedprotection" "test/testfiles/nonexistant_filename.txt"

	[ $status -eq 1 ]
	#assert_equal "$ACTUAL_RESULT" "$EXPECTED_OUTPUT"
}

@test "Tests whether an exception is thrown when computing the total nr of lines, if the file is not found ERRROR." {
	run_main_functions "custom_install_4_energizedprotection" "test/testfiles/nonexistant_filename.txt"

	[ $status -eq 0 ]
	#assert_equal "$ACTUAL_RESULT" "$EXPECTED_OUTPUT"
}

@test "Tests whether an exception is thrown when computing the total nr of lines, if the file is not found ERRROR." {
	! run_main_functions "custom_install_4_energizedprotection" "test/testfiles/nonexistant_filename.txt"

	[ $status -eq 0 ]
	#assert_equal "$ACTUAL_RESULT" "$EXPECTED_OUTPUT"
}

@test "Tests whether an exception is thrown when computing the total nr of lines, if the file is not found ERRROR." {
	! run_main_functions "custom_install_4_energizedprotection" "test/testfiles/nonexistant_filename.txt"
}


@test "Tests whether an exception is thrown when computing the total nr of lines, if the file is not found ERRROR." {
	run run_main_functions "custom_install_4_energizedprotection" "test/testfiles/nonexistant_filename.txt"
	expect_failure
}
 
@test "Tests whether an exception is thrown when computing the total nr of lines, if the file is not found ERRROR." {
	run run_main_functions "custom_install_4_energizedprotection" "test/testfiles/nonexistant_filename.txt"
	assert_failure
}
    
```
Which all result in failed tests.

**Comment by a-t-0** [comment]
Resolved in commit e70eb3c in branch https://github.com/HiveMinds-EU/Productivity-setup/tree/issue-59-Write_post-setup_tests_for_energized_protection_social_package .
